src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+        this microservice job is written is that the mapping is 1:1. There
+        isn't space for an empty line at the end of the file.
+        """
+        lines = self._count_lines(hashfile)
+        objects = self._count_files(objectsdir)
+        if lines == objects:
+            return True
+        self.job.pyprint(
+            "{}: Comparison failed with {} checksum lines and {} "
+            "transfer files".format(self.get_ext(hashfile), lines, objects),
+            file=sys.stderr,
+        )
+        return False
+
+    def compare_hashes(self, hashfile, objectsdir):
+        """Compare transfer files with the checksum file provided."""
+        if not self.count_and_compare_lines(hashfile, objectsdir):
+            return 1
+        try:
+            self._call("-c", "--strict", hashfile)
+            return 0
+        except subprocess.CalledProcessError as err:
+            if self.EXIT_NON_ZERO in str(err):
+                warn = "{}: comparison exited with status: {}. Please check the formatting of the checksums or integrity of the files.".format(
+                    self.get_ext(hashfile), err.returncode
+                )
+                self.job.pyprint(warn, file=sys.stderr)
+            for line in self._decode(err.output):
+                if not line:
+                    continue
+                if line.endswith(self.OKAY_STRING):
+                    continue
+                if (
+                    line.endswith(self.FAIL_STRING)
+                    or self.ZERO_STRING in line
+                    or self.IMPROPER_STRING in line
+                ):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(hashfile), line), file=sys.stderr
+                    )
+                if line.endswith(self.FAILED_OPEN):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(hashfile), line), file=sys.stderr
+                    )
+            return err.returncode
+
+    def version(self):
+        """Return version information for the command being called."""
+        try:
+            return self._call("--version")[0]
+        except subprocess.CalledProcessError:
+            return self.COMMAND
+
+    def get_command_detail(self):
+        """Provide some way for the user to get information out of the class to
+        write METS/PREMIS provenance information.
+        """
+        if not self.command_called:
+            raise PREMISFailure()
+        return 'program="{}"; version="{}"'.format(
+            " ".join(self.command_called), self.version()
+        )
+
+    @staticmethod
+    def _decode(out):
+        """Decode string output in Py2 or Py3 and return a list of lines to be
+        parsed elsewhere.
+        """
+        try:
+            return str(out, "utf8").split("\n")
+        except TypeError:
+            return out.decode("utf8").split("\n")
+
+    @staticmethod
+    def get_ext(path):
```

Comment:
If I understand the code paths correctly, `path` here is the same `path` the constructor received? If that's the case why is it necessary to pass it again and have this to be a `staticmethod`?

Reply:
In this instance I'm using them as helper functions so I call them within the class and outside of the class. So take for example this output I generate:
```python
        if os.path.exists(hashfile):
            try:
                hashsum = Hashsum(hashfile, job)
            except NoHashCommandAvailable:
                job.pyprint(
                    "Nothing to do for {}. No command available.".format(
                        Hashsum.get_ext(hashfile)
                    )
                )
                continue
```
If I can't initialize the object, I'd still like to return some information about what I was doing, so do it via this helper. I guess it's a bit of an unholy mix of OO and something else really? (Archivematica Microservice Style? :wink: ) I can rework it if there's a strong opinion either way.